src/usecases/counter/executor.rs

```diff
@@ -0,0 +1,88 @@
+use crate::bindings::WalletProvider;
+use crate::bindings::blssigcheckoperatorstateretriever::BLSSigCheckOperatorStateRetriever::getNonSignerStakesAndSignatureReturn;
+use crate::bindings::counter::{self, Counter};
+use crate::executor::interface::{ContractHandler, ExecutionResult};
+use alloy_primitives::{Bytes, FixedBytes};
+use anyhow::Result;
+use async_trait::async_trait;
+
+pub struct CounterHandler {
+    counter: Counter::CounterInstance<(), WalletProvider>,
+}
+
+impl CounterHandler {
+    pub fn new(counter: Counter::CounterInstance<(), WalletProvider>) -> Self {
+        Self { counter }
+    }
+
+    fn convert_non_signer_data(
```

Comment:
this should go into the bn254-el-solidity verifier section , it is not specifici to the counter